pkg/tickers/retry.go

```diff
@@ -0,0 +1,84 @@
+package tickers
+
+import (
+	"context"
+	"time"
+
+	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg"
+)
+
+// UpkeepPayload represents the payload for an upkeep task.
+//type UpkeepPayload struct {
+//	Hash string // Hash uniquely identifies the upkeep payload
+//}
+
+// CheckResult represents the result of a check pipeline.
+type CheckResult struct {
+	Payload   ocr2keepers.UpkeepPayload
+	Retryable bool
+	// ... other properties
+}
+
+// RetryTicker represents a ticker that handles retries.
+type RetryTicker struct {
+	*TimeTicker
+	nextRetries  map[time.Time]ocr2keepers.UpkeepPayload
+	newDurations map[string]time.Duration
+}
+
+// Retry adds a retryable result to the RetryTicker.
+func (rt *RetryTicker) Retry(result CheckResult) error {
+	payload := result.Payload
+	retryable := result.Retryable
+
+	if retryable {
```

Comment:
These can be combined into `if result.Retryable` without affecting code readability and it will be a minor simplification.

Reply:
if not retryable, throw an error